### ISSUE TITLE
ENH: Add join K nearest to sjoin_nearest

### ISSUE THREAD

**Issue by iki77** [geopandas/geopandas#2434]
#### Is your feature request related to a problem?

Currently, the sjoin_nearest function only allows spatial join to the 1st nearest geometry
I am unable to use the sjoin_nearest function to find up to k nearest geometry

#### Describe the solution you'd like

The sjoin_nearest should get a new parameter k_nearest so that it can be used to find up to k nearest geometry

**Comment by benoit9126** [comment]
I am also interested in such feature available from geopandas. 

As far as I understand, the `STRTree` C++ implementation of `geos` does not have a `k_nearest_neighbours` or an equivalent method. See [documentation](http://libgeos.org/doxygen/classgeos_1_1index_1_1strtree_1_1STRtree.html). Thus, the feature is not available in `pygeos` `STRTree` class and then not available directly in GeoPandas via the `PyGEOSSTRTreeIndex` class... I think we should first post an issue in the `geos` repository.

Note that the feature is available via the class `RTreeIndex` [here](https://github.com/geopandas/geopandas/blob/28462f8e7fd893e01be52f444e142a1a00c36f8e/geopandas/sindex.py#L536) using the `num_results` argument. Unfortunately, it will be removed.

**Comment by brendan-ward** [comment]
This is on hold until k nearest neighbor functionality is available in GEOS.  It is implemented in [JTS](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/index/strtree/STRtree.html) (from which many GEOS algorithms are derived), so it is primarily waiting on that to be ported from JTS to GEOS.
